README.md

```diff
@@ -1 +1,218 @@
+
+* `emitter.listeners(eventName)`: List the event listeners associated with the `eventName` event
+* `emitter.once(eventName, listener)`: Create event listener for `eventName` and execute it just one time. This is similar to `.on()` but fires only once.
+* `emitter.removeListener(eventName, listener)`: Remove event listener for `eventName`
+
+In our knock-knock example, we used the observer object `event`. It's not very interesting because it didn't do much. What's more interesting and useful is that we can extend Event Emitter properties and methods into ANY OBJECT/class. There's this convenient function `inherits()` from the core `util` module. It takes a child (`fancyChildObjectWithEvents`) and parent objects, and makes the child inherit from the parent (Event Emitter).
+
+```js
+var util = require('util')
+var events  = require('events')
+util.inherits(fancyChildObjectWithEvents, events.EventEmitter)
+```
+
+Imagine a situation when you're developing an email job module. This module needs to be flexible enough to send different notifications when the job is done: weekly, daily and monthly. 
+
+Event listeners can be located in different places than the class. This feature is useful for our email job class. We will create the class in `job.js`, then set up event listeners in `weekly.js`.
+
+The `job.js` has module imports:
+
+```js
+// job.js
+var util = require('util')
+var events = require('events')
+```
+
+Then create a class/object `Job` using pseudo-classical pattern. It must have `process` method which will launch the email job. Inside of that method, emit the `done` event when everything is finished - async! And one more thing, we can pass any data in the `emit` method. In this example, pass the time&date.
+
+```js
+var Job = function Job() {
+  var job = this // We'll use it later to pass 'this' to closures
+  // ...
+  job.process = function() {
+    // ...
+    setTimeout(function(){ // Emulate the delay of the job - async!
+      job.emit('done', { completedOn: new Date() })
+    }, 700)
+  }
+}
+```
+
+
+
+Now, extend the events and export the class:
+
+```js
+util.inherits(Job, events.EventEmitter)
+module.exports = Job
+```
+
+
+In the `weekly.js` file, you'll consume the `Job` and implement the custom event listener. In this case, it's a weekly email runner so the event listener will print `Weekly...`:
+
+```js
+// weekly.js
+var Job = require('./job.js')
+var job = new Job()
+
+job.on('done', function(details){
+  console.log('Weekly email job was completed at', details.completedOn)
+})
+
+job.process()
+```
+
+This is how you customize behavior of your classes by specifying event listeners later in different files! As mentioned before, the advantage of this over callbacks is that you can have multiple events, remove them, execute them once, etc. The reverse is possible as well.You can have an event listener in `job.js` and trigger the event in `weekly.js`. 
+
+```js
+// job.js
+// ...
+job.on('start', function(){
```

Comment:
We could even show students

```js
job.on('start', job.process)
```

since we don't need rely on context/scope inside the handler.